src/include/portable_endian.h

```diff
@@ -22,10 +22,27 @@
 
 #	include <endian.h>
 
-#elif defined(__linux__) || defined(__CYGWIN__)
+#elif defined(__linux__) && !defined(FOR_MANYLINUX) || defined(__CYGWIN__)
 
 #	include <endian.h>
 
+#elif defined(FOR_MANYLINUX)
```

Comment:
You can check for a glibc version. So, if you are Linux, but not glibc >= 2.9, then define these.

In `features.h` on glibc `__GLIBC__` and `__GLIBC_MINOR__` are defined.

So, something like (untested)

```c
#if (defined(__linux__) || defined(__CYGWIN__))
#if !defined(__GLIBC__) || __GLIBC__ >= 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 9)
#include <endian.h>
#else
// Definitions for old glibc versions
#endif
// ...
```